dot/network/service.go

```diff
@@ -576,6 +573,20 @@ func (s *Service) SendMessage(to peer.ID, msg NotificationsMessage) error {
 	return errors.New("message not supported by any notifications protocol")
 }
 
+func (s *Service) GetRequestResponseProtocol(subprotocol string, requestTimeout time.Duration,
+	maxResponseSize uint64) RequestResponseProtocol {
```

Comment:
Same here, we don't need to return an interface, we can return the actual implementation. ref: [accept-interfaces-return-structs-in-go](https://bryanftan.medium.com/accept-interfaces-return-structs-in-go-d4cab29a301b)

```suggestion
func (s *Service) GetRequestResponseProtocol(subprotocol string, requestTimeout time.Duration,
	maxResponseSize uint64) RequestResponseProtocol {
```